compiler/x/codegen/FPTreeEvaluator.cpp

```diff
@@ -1880,54 +1375,12 @@ TR::Register *OMR::X86::TreeEvaluator::fRegStoreEvaluator(TR::Node *node, TR::Co
    TR::Register *childGlobalReg = cg->machine()->getFPStackRegister(fpStackSlot);
    TR::Register *globalReg = cg->evaluate(child);
 
-   if (cg->useSSEForSinglePrecision())
-      {
-      if (globalReg->getKind() != TR_FPR)
-         globalReg = TR::TreeEvaluator::coerceFPRToXMMR(child, globalReg, cg);
-
-      machine->setXMMGlobalRegister(globalRegNum - machine->getNumGlobalGPRs(), globalReg);
-      cg->decReferenceCount(child);
-      return globalReg;
-      }
-   else
-      {
-      machine->setFPStackRegister(globalRegNum - machine->getNumGlobalGPRs(), toX86FPStackRegister(globalReg));
-
-      if (cg->decReferenceCount(child) == 0)
-         {
-         generateFPSTiST0RegRegInstruction(TR::InstOpCode::FSTRegReg, node, globalReg, globalReg, cg);
-         cg->stopUsingRegister(globalReg);
-         }
-      else if (childGlobalReg &&
-               childGlobalReg != globalReg)
-         {
-            {
-            int32_t i;
-            for (i=0;i<TR_X86FPStackRegister::NumRegisters;i++)
-               {
-               if (i != fpStackSlot)
-                  {
-                  TR::Register *reg = cg->machine()->getCopiedFPStackRegister(i);
-                  //if (!reg)
-                  //   dumpOptDetails("Global reg %d Stack slot %d iglobalreg %d reg address %p childreg %s\n", globalRegNum, fpStackSlot, i+machine->getNumGlobalGPRs(), reg, getDebug()->getName(childGlobalReg));
-                  //else
-                  //   dumpOptDetails("Global reg %d Stack slot %d iglobalreg %d reg %s childreg %s\n", globalRegNum, fpStackSlot, i+machine->getNumGlobalGPRs(), getDebug()->getName(reg), getDebug()->getName(childGlobalReg));
-
-                  if (reg == childGlobalReg)
-                     {
-                     TR::Register *popReg = cg->machine()->getFPStackRegister(i);
-                     generateFPSTiST0RegRegInstruction(TR::InstOpCode::FSTRegReg, node, childGlobalReg, childGlobalReg, cg, true);
-                     cg->stopUsingRegister(childGlobalReg);
-                     cg->machine()->getFPStackRegisterNode(fpStackSlot)->setRegister(popReg);
-                     break;
-                     }
-                  }
-               }
-            }
-         }
+   if (globalReg->getKind() != TR_FPR)
+      globalReg = TR::TreeEvaluator::coerceFPRToXMMR(child, globalReg, cg);
```

Comment:
Delete these two lines.  The globalReg can't be anything other than a TR_FPR.  You could add a TR_FATAL_ASSERT for this instead.